shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -1485,6 +1484,10 @@ private void sendViewportMetricsToFlutter() {
     viewportMetrics.devicePixelRatio = getResources().getDisplayMetrics().density;
     viewportMetrics.physicalTouchSlop = ViewConfiguration.get(getContext()).getScaledTouchSlop();
 
+    viewportMetrics.displayFeatures.clear();
```

Comment:
I also thought about separating the two lists within `ViewportMetrics`, and wouldn't be opposed.

I'm not too concerned about being required to remember to call clear on `viewportMetrics.displayFeatures`, though, because there the pr establishes a pattern that list is exclusively handled by this helper (`sendViewportMetricsToFlutter`) which takes care of clearing.

Reply:
Let us separate the lists in Viewport metrics then. I agree your helper does a good job of isolating the current logic but I think future logic is where we would get an accidental regression.

Reply:
I've separated the two lists, introduced a new `DisplayCutout` class, as that inherently represents less data than the generic `DisplayFeature`, and modified the view and renderer tests accordingly.